spec.html

```diff
@@ -25044,11 +25044,11 @@ <h1>decodeURI ( _encodedURI_ )</h1>
         <p>The `decodeURI` function is the <dfn>%decodeURI%</dfn> intrinsic object. When the `decodeURI` function is called with one argument _encodedURI_, the following steps are taken:</p>
         <emu-alg>
           1. Let _uriString_ be ? ToString(_encodedURI_).
-          1. Let _reservedURISet_ be a String containing one instance of each code unit valid in |uriReserved| plus `"#"`.
+          1. Let _reservedURISet_ be a String containing one instance of each code unit valid in |uriReserved| plus `#`.
           1. Return ? Decode(_uriString_, _reservedURISet_).
         </emu-alg>
         <emu-note>
-          <p>The code point `"#"` is not decoded from escape sequences even though it is not a reserved URI code point.</p>
+          <p>The code point `#` is not decoded from escape sequences even though it is not a reserved URI code point.</p>
```

Comment:
Seems like we should update `code point` → `code unit` here.

Reply:
It's a bit tricky. Although the input and output of `Decode` are just String values, they are different ways of encoding a ("true") URI consisting of Unicode code points. So it's somewhat ambiguous whether "decoded" in this sentence is talking about the whole effect of `Decode`, or a particular point within it at which an escape sequence could be (but isn't) decoded into a Unicode code point.

Given the placement of the note, it seems more plausible that it would be talking about the whole effect of `Decode`, rather than an internal detail, in which case changing the first occurrence of "code point" to "code unit" (as you suggest) would be appropriate. (Though rewording the sentence to not have that ambiguity would be a bigger improvement, I think.)

Reply:
Alternatively, maybe it's the parallel sentence below that should change in the opposite direction? Re-reading the sentence as a whole, it seems like it we could argue that it is making a note about the code point `#`, whereas the algorithm step is building a string from code units including **"#"**.

Reply:
> Alternatively, maybe it's the parallel sentence below that should change in the opposite direction?

That is, on line 25076, change "code unit" to "code point". 

> Re-reading the sentence as a whole, it seems we could argue that the note is purely about the code point `#`,

Yup, it's debatable.

> whereas the algorithm step is building a string from code units including **"#"**.

The algorithm is clearly building a string, although the phrase `each code unit valid in |uriReserved|` is slightly problematic, if we assume (as with the spec's other lexical grammars) that the terminal symbols are code points, not code units. In effect, the 'uri' grammar is not there to specify a syntax (nothing is parsed with `|uri|` as the goal symbol), but merely to define  `|uriReserved|` and `|uriUnescaped|` as sets of code units. But that's mostly beside the point.